networkx/convert.py

```diff
@@ -248,12 +248,91 @@ def to_dict_of_dicts(G, nodelist=None, edge_data=None):
     nodelist : list
        Use only nodes specified in nodelist
 
-    edge_data : list, optional
-       If provided,  the value of the dictionary will be
-       set to edge_data for all edges.  This is useful to make
-       an adjacency matrix type representation with 1 as the edge data.
-       If edgedata is None, the edgedata in G is used to fill the values.
-       If G is a multigraph, the edgedata is a dict for each pair (u,v).
+    edge_data : singleton, optional
+       If provided, the value of the dictionary will be set to `edge_data` for
+       all edges. Usual values could be `1` or `True`. If `edge_data` is
+       `None` (the default), the edgedata in `G` is used, resulting in a
+       dict-of-dict-of-dicts. If `G` is a MultiGraph, the result will be a
+       dict-of-dict-of-dict-of-dicts. See Notes for an approach to customize
+       handling edge data.
+
+    Returns
+    -------
+    dod : dict
+       A nested dictionary representation of `G`. Note that the level of
+       nesting depends on the type of `G` and the value of `edge_data`
+       (see Examples).
+
+    See Also
+    --------
+    from_dict_of_dicts, to_dict_of_lists
```

Comment:
Should these have rst style references? e.g. 

```
`func`:from_dict_of_dicts`
```

Reply:
They don't need them - the sphinx extensions that handles numpydoc-formatted docstring parsing (napoleon in this case) handles the roles automatically.

You can see the rendered documentation via the `circleci: build artifact` tab in the checks below. Here's a link to the page for `to_dict_of_dicts`: https://629-890377-gh.circle-artifacts.com/0/doc/build/html/reference/generated/networkx.convert.to_dict_of_dicts.html